webhooks/drivers.go

```diff
@@ -0,0 +1,106 @@
+package webhooks
+
+import (
+	"crypto/rsa"
+	"fmt"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher/v2"
+	"github.com/rancher/rancher-auth-service/util"
+	"github.com/urfave/cli"
+	"time"
+)
+
+//PrivateKey to be used for jwt creation
+var privateKey *rsa.PrivateKey
+
+func SetEnv(c *cli.Context) {
+	privateKeyFile := c.GlobalString("rsa-private-key-file")
+	if privateKeyFile != "" {
+		privateKey = util.ParsePrivateKey(privateKeyFile)
+	}
+}
+
+// ConstructPayload constructs jwt url
+func (d Driver) ConstructPayload(input map[string]interface{}) (map[string]interface{}, error) {
+	payload := make(map[string]interface{})
+	jwt, err := util.CreateTokenWithPayload(input, privateKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating JWT")
+	}
+	payload["url"] = jwt
+	return payload, nil
+}
+
+//Process does a lookup for driver and calls it
+func (d Driver) Execute(payload map[string]interface{}) (int, map[string]interface{}, error) {
+	jwtSigned := payload["url"]
+	token, err := jwt.Parse(jwtSigned.(string), func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return privateKey, nil
+	})
+
+	if claims, ok := token.Claims.(jwt.MapClaims); ok {
+		d.DriverFunction, err = LookUpDriver(claims["driver"].(string))
```

Comment:
This will panic if driver is not present or not a string. Protect against that. This comment applies across the board.